ddtrace/opentelemetry/tracer.go

```diff
@@ -92,6 +94,69 @@ func (t *oteltracer) Start(ctx context.Context, spanName string, opts ...oteltra
 	// we have to record the attributes  locally.
 	// The span operation name will be calculated when it's ended.
 	s := tracer.StartSpan(spanName, ddopts...)
+
+	// Get the current OpenTelemetry baggage.
+	otelBag := otelbaggage.FromContext(ctx)
+	// Get the ddtrace baggage as a map[string]string.
+	ddBag := baggage.All(ctx)
+
+	// Merge the two baggage maps.
+	// If there are conflicts, the OpenTelemetry baggage wins.
+	var mergedBag otelbaggage.Baggage
+	switch {
```

Comment:
Nice! Can we refactor some of the logic in this switch statement into helper functions? This would:
- clean up the code inside of tracer.Start, which is already a massive function
- remove the need for so many in-line comments, as the helper function names would be sufficiently descriptive to showcase what you're doing, and why.
You could even extract the baggage logic (starting at line 98)out into its own helper function and use sub-"helper"-functions inside of that.